pkg/handlers/archive.go

```diff
@@ -101,12 +101,13 @@ func (h *archiveHandler) HandleFile(ctx logContext.Context, input fileReader) ch
 var ErrMaxDepthReached = errors.New("max archive depth reached")
 
 // openArchive recursively extracts content from an archive up to a maximum depth, handling nested archives if necessary.
-// It takes a reader from which it attempts to identify and process the archive format. Depending on the archive type,
-// it either decompresses or extracts the contents directly, sending data to the provided channel.
+// It takes a string representing the path to the archive and a reader from which it attempts to identify and process the archive format.
+// Depending on the archive type, it either decompresses or extracts the contents directly, sending data to the provided channel.
 // Returns an error if the archive cannot be processed due to issues like exceeding maximum depth or unsupported formats.
 func (h *archiveHandler) openArchive(
 	ctx logContext.Context,
 	depth int,
+	archiveEntryPath string,
```

Comment:
I think these could be combined into a single variable that's a slice of the paths. For example: `["foo", "bar"]` has `depth = 2` and `archiveEntryPath = foo/bar`. Is there a reason I'm not seeing to keep them as two distinct variables?

Reply:
That's probably a cleaner way to implement it. And for the case of decompression, just add in `["foo", ".gz decompression", "bar"]`?